agent/pkg/singularity/singularity.go

```diff
@@ -81,14 +82,22 @@ func New(opts options.Options) (*SingularityClient, error) {
 		return nil, fmt.Errorf("preparing agent tmp: %w", err)
 	}
 
+	// Validate image root (if provided)
+	if opts.ImageRoot != "" {
+		if _, err := os.ReadDir(opts.ImageRoot); err != nil {
+			return nil, fmt.Errorf("reading image root directory: %w", err)
```

Comment:
Do we want to show the value of "opts.ImageRoot" in the error message?

Reply:
In testing I have done, the path appears in the error message (see in the PR description for an example).